dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/VisitorLitmusAArch64.java

```diff
@@ -53,9 +57,32 @@ public Object visitMain(LitmusAArch64Parser.MainContext ctx) {
         visitVariableDeclaratorList(ctx.variableDeclaratorList());
         visitInstructionList(ctx.program().instructionList());
         VisitorLitmusAssertions.parseAssertions(programBuilder, ctx.assertionList(), ctx.assertionFilter());
-        return programBuilder.build();
+        Program prog = programBuilder.build();
+        replaceXZRRegister(prog);
+
+        return prog;
+
     }
 
+    /*
+    The "xzr" register plays a special role in AArhc64:
+      1. Reading accesses always return the value 0.
+      2. Discards data when written.
+     TODO: The below code is a simple fix to guarantee point 1. above.
+      Point 2. might also be resolved: although we do not prevent writing to xzr,
+      the value of xzr is never read after the transformation so its value is effectively 0.
+      However, the exists/forall clauses could still refer to that register and observe a non-zero value.
+     */
+    private void replaceXZRRegister(Program program) {
+        final ExpressionVisitor<Expression> xzrReplacer = new ExprTransformer() {
+            @Override
+            public Expression visitRegister(Register reg) {
+                return reg.getName().equals("xzr") ? expressions.makeGeneralZero(reg.getType()) : reg;
+            }
+        };
+        program.getThreadEvents(RegReader.class)
+                .forEach(e -> e.transformExpressions(xzrReplacer));
+    }
```

Comment:
I think RISCV has the same situation (and maybe even Power?), just with a different register name.
Maybe it makes sense to put this functionality into `ProgramBuilder` with an additional `Register` parameter for the replacement. Then we can reuse that function.

Reply:
Done

Reply:
I don't like your solution. Specifying the 0-register on construction of `ProgramBuilder` does nothing but introduce unnecessary state. Also, as soon as there are multiple such registers to replace (e.g. of different sizes), the solution won't work.
Just make the replacement function public and let it take the register name as input: it's less code, less state, and more general.
The litmus parser should just make sure to call that function right before `.build()`.